src/reconcile/rec_write.c

```diff
@@ -1987,10 +1987,9 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
 		 * have one.  If we don't have room for another split chunk,
 		 * add whatever space remains in this page.
 		 */
-		if (inuse + r->split_size <= r->page_size)
-			r->space_avail =
-			    r->split_size - WT_PAGE_HEADER_BYTE_SIZE(btree);
-		else {
+		r->space_avail =
+		    r->split_size - WT_PAGE_HEADER_BYTE_SIZE(btree);
+		if (inuse + r->space_avail > r->page_size) {
```

Comment:
In case it isn't clear (it wasn't to me!), previously we could take the `else` here and calculate a value larger than `split_size - header`.  That then caused problems later.  Now we only use the `page_size` calculation if it would definitely overflow to add `split_size`, taking the header into account.